src/renderer/store/tachyon.store.ts

```diff
@@ -40,6 +67,221 @@ async function connect() {
+                userSubList.push(tachyonStore.activeLobby.members[memberKey].id);
+            }
+        }
+        window.tachyon.request("user/subscribeUpdates", { userIds: userSubList });
+    } catch (error) {
+        console.error("Error with request lobby/join", error);
+        tachyonStore.error = "Error with request lobby/join";
+    }
+}
+
+// We use this function to normalize both LobbyCreateRequestData and LobbyJoinRequestData into the Lobby type for use in the renderer
+function parseLobbyResponseData(data: LobbyCreateOkResponseData | LobbyJoinOkResponseData) {
+    //Do some cleanup in case there's old data in the stores.
+    battleStore.battleOptions.mapOptions.customStartBoxes = [];
+
+    //Set up a basic object to hold the data
+    const lobbyObject: Lobby = {
+        id: data.id,
+        name: data.name,
+        playerCount: 0, //Placeholder value; We can count this from the entries below in the team config
+        maxPlayerCount: 0, //Placeholder value; Ditto for calculating max players
+        mapName: data.mapName,
+        engineVersion: data.engineVersion,
+        gameVersion: data.gameVersion,
+        allyTeams: {},
+        members: {},
+    };
+    battleStore.battleOptions.engineVersion = data.engineVersion;
+    battleStore.battleOptions.gameVersion = data.gameVersion;
+    db.maps.get(data.mapName).then((map) => {
+        battleStore.battleOptions.map = map;
+    });
+    for (const allyKey in data.allyTeamConfig) {
+        const item = data.allyTeamConfig[allyKey];
+        lobbyObject.allyTeams![allyKey] = {
+            startBox: data.allyTeamConfig[allyKey].startBox,
+            maxTeams: item.maxTeams,
+            teams: {},
+        };
+        for (const teamKey in item.teams) {
+            const team = item.teams[teamKey];
+            lobbyObject.allyTeams![allyKey].teams![teamKey] = { maxPlayers: team.maxPlayers };
+            lobbyObject.maxPlayerCount += team.maxPlayers; //Lobby maximum is sum of all AllyTeams>Teams>MaxPlayers
+        }
+        //Here we assign the startbox for the AllyTeam to the battlestore so they match what we set when the lobby was created.
+        battleStore.battleOptions.mapOptions.customStartBoxes.push(data.allyTeamConfig[allyKey].startBox);
+    }
+    //FIXME: Spectators need to be separated once implemented
+    for (const memberKey in data.members) {
+        lobbyObject.playerCount++; //Increment 1 for each player already in the lobby when created/joined.
+        const member = data.members[memberKey];
+        lobbyObject.members![memberKey] = {
+            id: member.id,
+            allyTeam: member.allyTeam,
+            team: member.team,
+            player: member.player,
+            type: "player",
+        };
+    }
+    return lobbyObject;
+}
+
+async function leaveLobby() {
+    try {
+        tachyonStore.error = undefined;
+        const response = await window.tachyon.request("lobby/leave");
```

Comment:
The ``lobby/leave`` request is basically always successful or it fails for reasons that you're not in a lobby anyway. If, for some reason, we have to clear the lobby data _but_ cannot get a success response here, the code below will not clear the status. Move it up so it happens either outside of the try/catch or before the request is sent.